app/javascript/lib/Entities/Board/Province.js

```diff
@@ -0,0 +1,65 @@
+import Entity from '../Entity';
+
+export default class Province extends Entity {
+  constructor(isLand, representation) {
+    super(representation);
+
+    this.isLand = isLand;
+
+    this.ownership = null;
+    this.hasArmsFactory = false;
+    this.hasNavalFactory = false;
+
+    this.friendlyUnits = new Map();
+    this.hostileUnits = new Map();
+  }
+
+  get isOcean() {
+    return this.isLand === false;
+  }
+  set isOcean(bool) {
+    this.isLand = bool === false;
+  }
+
+  get hasFactory() {
+    return this.hasArmsFactory || this.hasNavalFactory;
+  }
+  get isDevoidOfFactory() {
+    return this.hasFactory === false;
+  }
+
+  get isOccupiedByHostiles() {
+    return this.hostileUnits.size > 0;
+  }
+  get isNotOccupiedByHostiles() {
+    return this.isOccupiedByHostiles === false;
+  }
+}
+
+export const translateProvinceModel = (representation, allProvinces, allUnits, gameBoard) => {
+  // temporary until migration is complete
```

Comment:
Can you elaborate a little bit on how this will look when the migration is complete? Are we not going to have to pass all provinces, all units and the game board in each time?